lib/Sema/CSSimplify.cpp

```diff
@@ -3053,17 +3053,17 @@ getArgumentLabels(ConstraintSystem &cs, ConstraintLocatorBuilder locator) {
 /// conformances, but this is fine because it doesn't get invoked in the normal
 /// name lookup path (only when lookup is about to fail).
 static bool hasDynamicMemberLookupAttribute(CanType ty,
```

Comment:
Deduping this method with `getDynamicCallableMethods` (which also defines the same "type traversal" logic) would be nice. However, the two methods use different cache types (`llvm::DenseMap<CanType, bool>` vs `llvm::DenseMap<CanType, DynamicCallableMethods>`), which makes generalization difficult. 🙁

Any generalization ideas appreciated.